admin/billing/biller.go

```diff
@@ -0,0 +1,101 @@
+package billing
+
+import (
+	"context"
+	"time"
+
+	"github.com/rilldata/rill/admin/database"
+)
+
+const (
+	DefaultPlanID   = "Starter"
+	SupportEmail    = "support@rilldata.com"
+	DefaultTimeZone = "UTC"
+)
+
+type Biller interface {
+	Name() string
+	GetDefaultPlan(ctx context.Context) (*Plan, error)
+	GetPlans(ctx context.Context) ([]*Plan, error)
+	// GetPublicPlans for listing purposes
+	GetPublicPlans(ctx context.Context) ([]*Plan, error)
+	// GetPlan returns the plan with the given Rill plan ID or biller plan ID.
+	GetPlan(ctx context.Context, rillPlanID string, billerPlanID string) (*Plan, error)
+
+	// CreateCustomer creates a customer for the given organization in the billing system and returns the external customer ID.
+	CreateCustomer(ctx context.Context, organization *database.Organization) (string, error)
+
+	// CreateSubscription creates a subscription for the given organization.
+	// The subscription starts immediately.
+	CreateSubscription(ctx context.Context, customerID string, plan *Plan) (*Subscription, error)
+	CancelSubscription(ctx context.Context, subscriptionID string, cancelOption SubscriptionCancellationOption) error
+	GetSubscriptionsForCustomer(ctx context.Context, customerID string) ([]*Subscription, error)
+	ChangeSubscriptionPlan(ctx context.Context, subscriptionID string, plan *Plan) (*Subscription, error)
+	// CancelSubscriptionsForCustomer deletes the subscription for the given organization.
+	// cancellationDate only applicable if option is SubscriptionCancellationOptionRequestedDate
+	CancelSubscriptionsForCustomer(ctx context.Context, customerID string, cancelOption SubscriptionCancellationOption) error
+
+	ReportUsage(ctx context.Context, customerID string, usage []*Usage) error
+
+	GetReportingGranularity() UsageReportingGranularity
+	GetReportingWorkerCron() string
+}
+
+type Plan struct {
+	BillerID          string // ID of the plan in the external billing system
+	RillID            string // ID of the plan in Rill, can be empty if biller does not support it
+	Name              string
```

Comment:
I would suggest aligning the terminology here with the one we have for orgs, projects, etc. – i.e. using "ID" for non-user facing IDs, "name" for unique user-facing names (e.g. for use in the CLI or URL slugs), and "display name" for friendly non-unique user facing names (like UI titles).

```go
type Plan struct {
  ID string // ID from Orb, e.g. "6753a734-e327-43c4-a46a-55f8d00721f5"
  Name string // Unique name we assign, e.g. "starter-plan"
  DisplayName string // Friendly name, e.g. "Starter Plan"
  ...
}